CMakeLists.txt

```diff
@@ -1099,27 +1125,33 @@ ENDIF(PCRE2_BUILD_TESTS)
 # Installation
 
 SET(CMAKE_INSTALL_ALWAYS 1)
+set(TARGETS_EXPORT_NAME pcre2-targets)
 
 INSTALL(TARGETS ${targets}
-        RUNTIME DESTINATION bin
+        EXPORT ${TARGETS_EXPORT_NAME}
+        PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+INSTALL(EXPORT ${TARGETS_EXPORT_NAME}
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/pcre2
+        NAMESPACE pcre2::
+        FILE pcre2-targets.cmake)
 INSTALL(FILES ${pkg_config_files} DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 INSTALL(FILES "${CMAKE_CURRENT_BINARY_DIR}/pcre2-config"
   DESTINATION bin
   # Set 0755 permissions
   PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
 
-INSTALL(FILES ${PCRE2_HEADERS} ${PCRE2POSIX_HEADERS} DESTINATION include)
-
 # CMake config files.
 set(PCRE2_CONFIG_IN  ${CMAKE_CURRENT_SOURCE_DIR}/cmake/pcre2-config.cmake.in)
 set(PCRE2_CONFIG_OUT ${CMAKE_CURRENT_BINARY_DIR}/cmake/pcre2-config.cmake)
-configure_file(${PCRE2_CONFIG_IN} ${PCRE2_CONFIG_OUT} @ONLY)
-set(PCRE2_CONFIG_VERSION_IN  ${CMAKE_CURRENT_SOURCE_DIR}/cmake/pcre2-config-version.cmake.in)
+configure_package_config_file(${PCRE2_CONFIG_IN} ${PCRE2_CONFIG_OUT} INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/pcre2)
 set(PCRE2_CONFIG_VERSION_OUT ${CMAKE_CURRENT_BINARY_DIR}/cmake/pcre2-config-version.cmake)
-configure_file(${PCRE2_CONFIG_VERSION_IN} ${PCRE2_CONFIG_VERSION_OUT} @ONLY)
-install(FILES ${PCRE2_CONFIG_OUT} ${PCRE2_CONFIG_VERSION_OUT} DESTINATION cmake)
+write_basic_package_version_file(${PCRE2_CONFIG_VERSION_OUT}
+                                 VERSION ${PCRE2_MAJOR}.${PCRE2_MINOR}.0
+                                 COMPATIBILITY SameMajorVersion)
+install(FILES ${PCRE2_CONFIG_OUT} ${PCRE2_CONFIG_VERSION_OUT} DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/pcre2)
```

Comment:
could you verify that it doesn't have an impact?, is there a minimum cmake version that will be needed to include those extra paths?

no idea why it was set that way originally but the change is somehow recent with 2410fbe3869cab403f02b94caa9ab37ee9f5854b by [jwsblokland](https://github.com/jwsblokland)

Reply:
The earliest CMake version I found that specifies this path for both Windows and Unix [is 3.7](https://cmake.org/cmake/help/v3.7/command/find_package.html#config-mode-search-procedure:~:text=%3Cprefix%3E/%3Cname%3E*/(lib/%3Carch%3E%7Clib%7Cshare)/%3Cname%3E*/(cmake%7CCMake)/%20%20(W/U)). For reference the minimum CMake version that Google supports [is 3.13](https://github.com/google/oss-policies-info/blob/main/foundational-cxx-support-matrix.md) ([the earliest version that ships in a supported Linux distribution](https://opensource.google/documentation/policies/cplusplus-support#build_systems:~:text=We%20support%20the%20oldest%20release%20that%20comes%20standard%20with%20a%20supported%20distro.)).

Reply:
FWIW, I am not advocating to use cmake 3.3 forever (indeed, if you look at the git log, I'd been quietly rising it from 3.0, as new features were used), just making sure that if we are going to say that is the minimum version supported, it actually is.

One interesting thing I noticed after installing cmake [3.3.2](https://cmake.org/files/v3.3/) and doing a deployment is that the original code actually puts those 2 files in `${INSTALL_PREFIX}/cmake`, so I wouldn't be surprised if whoever is using this might be adding that special path somehow already and therefore will get a nasty surprise when it goes away.

I am testing in a Fedora 38 system, so maybe it is different in Windows and they might be using that and it might work based on your reading of the documentation, but either way it is a regression.

Reply:
Where can we document this change?

Reply:
Ideally, the code will be changed to be backward compatible so there is no need to document it.

Otherwise, we need to make @PhilipHazel aware so it is mentioned in the ChangeLog and he includes a note of it for the next release announcement.  It might even force us to do first a RC release since whoever might be affected will need to make sure to update their code first.

In this case I am sympathetic to the change, since it is likely to work by default and is more consistent, although it might require setting `FIND_LIBRARY_USE_LIB64_PATHS` in some cases.

Reply:
You are discussing the install location of the CMake config file, which is the key to locating the package.
For sane values of `CMAKE_INSTALL_LIBDIR` in relation to the prefix, both the old and the new location is searched by [`find_package`](https://cmake.org/cmake/help/v3.3/command/find_package.html) Config mode, also in version 3.3, and regardless of operating system:
The old location matches `<prefix>/(cmake|CMake)/`.
The new location matches `<prefix>/(lib/<arch>|lib|share)/cmake/<name>*/`.

Given that exported configs usually consist of several interacting files (config, version, targets, per-config details), this change is necessary and expected, in order to organize the files in a local unit.

Reply:
I agree the change seems to be in the right direction, but we are likely going to need something to put in the release announcement so that anyone that might be affected, is aware of the change and can do whatever change might be needed on their side.

Our old code, wasn't very friendly to multiarch systems or cross compilation because it was installed in such a generic location, but that path was specifically chosen for some reason I am not aware of, so the least we could do is check with the author and come with some scenarios for a migration (if needed).